docs/graph-algorithms/breadth-first-search.mdx

```diff
@@ -0,0 +1,416 @@
+    {
+        v1 = Q[++front];
+        printf("%d\n", v1);
+        for (v2 = 0; v2 < n; v2++)
+        {
+            if (g[v1][v2] == TRUE && visit[v2] == FALSE)
+            {
+                Q[++rear] = v2;
+                visit[v2] = TRUE;
+            }
+        }
+    }
+}
+
+
+/*
+Sample Case : 
+This is a Program To Create a Graph
+The Display Is In Breadth First Manner
+Enter no. of nodes 4
+Enter the vertices no. starting from 0
+Enter the vertices v1 & v2
+0 1
+Add more edges??(y/n)y
+Enter the vertices v1 & v2
+0 2
+Add more edges??(y/n)y
+Enter the vertices v1 & v2
+1 3
+Add more edges??(y/n)y
+Enter the vertices v1 & v2
+2 3
+Add more edges??(y/n)
+The Adjacency Matrix for the graph is
+0 1 1 0
+1 0 0 1
+1 0 0 1
+0 1 1 0
+Enter the Vertex from which you want to traverse : 0
+The Breadth First Search of the Graph is
+0 
+1 
+2 
+3
+Do you want to traverse from any other node?
+Enter the Vertex from which you want to traverse 1
+The Breadth First Search of the Graph is
+1 
+0 
+3 
+2
+Do you want to traverse from any other node?
+*/
+
+```
+
+```java
+
+import java.util.*;
+
+public class BFS {
+
+	public static void breadthFirst(int edges[][],int V) {
+//		A visited array to mark if the node is visited then not to go there again
+		boolean visited[]=new boolean[V];
+		
+		for(int i=0;i<V;i++) {
+			if(!visited[i]) {
+				breadthFirstHelper(edges,visited,i);
+			}
+		}
+	}
+	
+	public static void breadthFirstHelper(int edges[][],boolean visited[],int sv) {
+		
+		int V=edges.length;
+		
+		visited[sv]=true;
+		
+		Queue<Integer> queue=new LinkedList<Integer>();
+		queue.add(sv);
+		
+		while(!queue.isEmpty()) {
+			
+			int size=queue.size();
+			
+			for(int i=0;i<size;i++) {
+				int frontNode=queue.poll();
+				for(int j=0;j<V;j++) {
+					if(edges[frontNode][j]==1&&!visited[j]) {
+						queue.add(j);
+						visited[j]=true;
+					}
+				}
+				System.out.print(frontNode+" ");
+			}
+		}
+		
+	}
+	
+	public static void main(String[] args) {
+		
+		Scanner sc=new Scanner(System.in);
+	
+		System.out.println("Enter the number of vertices in Graph : ");
+		int V=sc.nextInt();
+		
+		System.out.println("Enter the number of edges in the Graph : ");
+		int E=sc.nextInt();
+		
+//		Adjaceny Matrix :
+		int edges[][]=new int[V][V];
+		
+//		Taking edges as input :
+		System.out.println("Enter the edges :");
+		for(int i=0;i<E;i++) {
+			int sv=sc.nextInt();
+			int ev=sc.nextInt();
+			edges[sv][ev]=1;
+			edges[ev][sv]=1;
+		}
+		
+		System.out.println("BFS Traversal : ");
+		breadthFirst(edges,V);
+
+	}
+	
+/*
+Sample Case : 
+Enter the number of vertices in Graph : 
+4
+Enter the number of edges in the Graph : 
+4
+Enter the edges :
+0 1
+1 2
+0 2
+2 3
+BFS Traversal : 
+0 1 2 3 
+*/
+	 
+}
+
+```
+
+## Complexity
+- Time Complexity: O(V+E) V: Vertices  E: Edges
+
+## Credits
+- [Neo Algo Project C++](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/graphs/Best_First_Search_Algorithm.cpp)
```

Comment:
I will leave a few examples of how credits are given 
[Insertion Sort](https://neoalgo.tesseractcoding.com/docs/sorting-algorithm/insertion-sort#credits)
[Merge Sort](https://neoalgo.tesseractcoding.com/docs/sorting-algorithms/merge-sort#credits)
[Quick Sort](https://neoalgo.tesseractcoding.com/docs/sorting-algorithms/Quick-sort#credits)
While the way you have written is not wrong it doesn't credit the original authors that much. And good work on the article. Thank you for taking out time and making this article.

Reply:
@gat786 Done 
Thank you for your Good Words after this one I will work on DFS as soon as possible